public/main/plugins/coincap/index.js

```diff
@@ -56,15 +54,20 @@ function start (pluginEmitter) {
 }
 
 function sendCachedPrice (webContents) {
-  getDb().collection('state')
-    .findOneAsync({ type: 'coincap-eth-usd' })
-    .then(function (doc) {
-      if (!doc) {
-        return
+  getEthPrice()
+    .then(function (price) {
+      if (!price) {
+        logger.debug('Getting ETH price')
+        return coincap.coin('ETH')
+          .then(res => res.price)
       }
 
-      logger.debug('Sending cached ETH price')
-      emitPrice(webContents, doc.price)
+      logger.debug('Using cached ETH price')
+      return price
+    })
+    .then(function (price) {
+      logger.debug('Sending ETH price')
+      emitAndCachePrice(webContents, price)
```

Comment:
Maybe we can move this to line `62` and chain the `.then` to cache the price only when we retrieve the price from `coincap`. Or the event emission (`emitAndCachePrice`) is necessary for both scenarios?

Reply:
@ndelvalle needed in both cases: we try to get the price from cache, if not there, we get it from CoinCap and in either case, we emit the event and store in cache. The second store is redundant but simplifies the logic and does no harm.